File: backend/Dtos/ItemPhotoDto.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Dtos
{
    public class ItemPhotoDto
    {
        public int Id { get; set; }

        public string PhotoUrl { get; set; } = string.Empty;

        public bool IsPrimary { get; set; } = false;

        public int DisplayOrder { get; set; } = 0;

        public DateTime UploadedAt { get; set; }
    }

    public class AddItemPhotoDto
    {
        [Required]
        public string PhotoUrl { get; set; } = string.Empty;
        public bool IsPrimary { get; set; } = false;
        public int DisplayOrder { get; set; } = 0;
    }

    public class UpdateItemPhotoDto
    {
        [Required]
        public int Id { get; set; }
        public bool? IsPrimary { get; set; }
        public int? DisplayOrder { get; set; }
    }

    public class ReorderItemPhotosDto
    {
        [Required]
        public int ItemId { get; set; }

        [Required]
        public List<int> PhotoIdsInOrder { get; set; } = new();
    }

    public class SetPrimaryPhotoDto
    {
        [Required]
        public int PhotoId { get; set; }

        [Required]
        public int ItemId { get; set; }
    }

    public class PhotoOrderUpdateDto
    {
        [Required]
        public int PhotoId { get; set; }

        [Required]
        public int DisplayOrder { get; set; }
    }

    public class DeleteItemPhotoDto
    {
        [Required(ErrorMessage = "Photo ID is required")]
        public int PhotoId { get; set; }

        [Required(ErrorMessage = "Item ID is required")]
        public int ItemId { get; set; }
    }

}

File: frontend/src/app/dtos/itemPhotoDto.ts
export interface ItemPhotoDto {
  id: number;
  photoUrl: string;
  isPrimary: boolean;
  displayOrder: number;
  uploadedAt: string;
}

export interface AddItemPhotoDto {
  photoUrl: string;
  isPrimary?: boolean;
  displayOrder?: number;
}

export interface UpdateItemPhotoDto {
  id: number;
  isPrimary?: boolean | null;
  displayOrder?: number | null;
}

export interface ReorderItemPhotosDto {
  itemId: number;
  photoIdsInOrder: number[];
}

export interface SetPrimaryPhotoDto {
  photoId: number;
  itemId: number;
}

export interface PhotoOrderUpdateDto {
  photoId: number;
  displayOrder: number;
}

export interface DeleteItemPhotoDto {
  photoId: number;
  itemId: number;
}